spec/integration/show_spec.rb

```diff
@@ -0,0 +1,46 @@
+require 'capybara/rails'
+require 'capybara/minitest'
+
+RSpec.describe 'User show page', type: :system do
+  let!(:user) do
+    User.create(name: 'John', photo: 'https://example.com/profile.jpg', bio: 'Software Engineer')
+    User.create(name: 'Daniel', photo: 'https://example.com/profile.jpg', bio: 'Electrical Engineer')
+  end
+
+  let!(:posts) do
+    user.posts.create([
+                        { text: 'First post', comments_counter: 2, likes_counter: 5 },
+                        { text: 'Second post', comments_counter: 3, likes_counter: 7 }
+                      ])
+  end
+
+  it 'displays user information and posts' do
+    visit user_path(user)
+    expect(page).to have_content('John')
+    expect(page).to have_content("Number of post: #{user.posts_counter}")
+
+    expect(page).to have_content('Bio :')
+    expect(page).to have_content('Software Engineer')
+  end
+
+  it 'displays user information and posts' do
+    visit user_path(user)
+    expect(page).to have_content('Daniel')
+    expect(page).to have_content("Number of post: #{user.posts_counter}")
+
+    expect(page).to have_content('Bio :')
+    expect(page).to have_content('Electrical Engineer')
+  end
+
+  it 'displays user profile' do
+    visit user_path(user)
+
+    expect(page).to have_css('.user-profile img[src*="https://example.com/profile.jpg"]')
+  end
+
+  it 'displays user posts path' do
+    visit user_path(user)
+
+    expect(page).to have_link('See all posts', href: user_posts_path(user), class: 'all_posts')
+  end
```

Comment:
- [ ] Kindly write the following test scenarios for the `User#Show` page based on this project's requirements 😎 💻:

- Test that asserts you can see the user's first 3 posts. (for this, I would advise you to create four posts for a user in the test file and test if the recent posts are being rendered)
- Test that redirects to the post's show page when you click on a user's post.
- Test that redirects to the user's post's index page when you click on the see all posts button.